src/api/gitHub.js

```diff
@@ -290,11 +221,114 @@ export const determineGraphQLEndpoint = async (server) => {
   }
 };
 
+
+/**
+ * fetch options for fetchRepoInOrg
+ *
+ * @param {string} org the organization
+ * @param {string} token the token
+ * @param {string} cursor the last repository fetched
+ * @param {boolean} selfsigned the selfsigned option 
+ * @returns {object} the fetch options
+ */
+export const fetchRepoInOrgInfoOptions = async (org, token, cursor, selfsigned) => {
+  let fetchOptions = {
+    method: "POST",
+    headers: {
+      Authorization: `bearer ${token}`,
+    },
+    body: JSON.stringify({
+      query: `{
+        rateLimit {
+          limit
+          cost
+          remaining
+          resetAt
+        }
+        organization(login: "${org}") {
+          repositories(first: 50${cursor}){
+            totalCount
+            edges {
+              cursor
+              node {
+                projects(first:1){
+                  totalCount
+                }  
+                hasWikiEnabled
+                issues(first: 1) {
+                  totalCount
+                }
+                pullRequests(first: 1) {
+                  totalCount
+                }
+                name
+                id
+                url
+                isPrivate
+              }
+            }
+          }
+        }
+      }`,
+    }),
+  }
+  if (selfsigned) {
+    /**
+    * set agent to a new https.Agent if it hasn't alredy been set
+    */
+    if (Object.entries(agent).length === 0) {
+      agent = new https.Agent({rejectUnauthorized: false,});
+    }
+    fetchOptions.agent = agent
```

Comment:
We can simplify this code significantly and avoid having the `agent` variable defined at the top level of this file:

```suggestion
    fetchOptions.agent = new https.Agent({ rejectUnauthorized: false });
```

Reply:
I changed the logic as suggested.  

I had considered this simpler approach.  I chose to introduce the complexity of using the top level ```agent``` variable so we can optimize the application.  Using the top level ```agent``` variable gives us the ability to use a single instance of https.Agent.  I thought that this was important because ```fetchRepoInOrg``` is called from ```fetchRepoMetrics``` which is called recursively to page through the repos 50 at a time.   When```fetchOrgInfoOptions``` and ``` fetchRepoInOrgInfoOptions``` were called the logic checks if the top level ```agent``` variable was already set, if it is already set then the top level ```agent``` variable was used as the agent in the fetch options, otherwise top level ```agent```  will be set to a new instance ```new https.Agent``` only once.

Reply:
Makes sense 👍 I understand your thinking now. I did some quick benchmarking and calling `new https.Agent` has a very low overhead, so I think this is fine.